patterns/multisig/VirtualMultisigRune.ts

```diff
@@ -0,0 +1,244 @@
+      .map(([value, memberProxies]) =>
+        Rune.array(memberProxies.map((memberProxy) =>
+          chain.extrinsic({
+            type: "Balances",
+            value: {
+              type: "transfer",
+              dest: MultiAddress.Id(memberProxy),
+              value,
+            },
+          })
+        ))
+      )
+      .into(MetaRune)
+      .flat()
+    const multisig = Rune
+      .rec({
+        signatories: memberProxies,
+        threshold,
+      })
+      .into(MultisigRune, chain)
+    const multisigExistentialDepositCall = chain.extrinsic(Rune.rec({
+      type: "Balances",
+      value: Rune.rec({
+        type: "transfer",
+        dest: multisig.accountId.map(MultiAddress.Id),
+        value: deposits,
+      }),
+    }))
+    const existentialDepositCalls = Rune
+      .tuple([memberProxyExistentialDepositCalls, multisigExistentialDepositCall])
+      .map(([m, s]) => [...m, s])
+    const existentialDeposits = chain
+      .extrinsic(Rune.rec({
+        type: "Utility",
+        value: Rune.rec({
+          type: "batchAll",
+          calls: existentialDepositCalls,
+        }),
+      }))
+      .signed({ sender: configurator })
+      .sent()
+      .dbgStatus("Existential deposits:")
+      .finalized()
+
+    const ownershipSwapCalls = Rune
+      .tuple([configurator, members, memberProxies, stashProxy, multisig.address])
+      .map(([configurator, members, memberProxies, stashProxy, multisigAddress]) =>
+        Rune.array([
+          ...replaceDelegateCalls(
+            chain,
+            MultiAddress.Id(stashProxy),
+            configurator.address,
+            multisigAddress,
+          ),
+          ...memberProxies.flatMap((proxy, i) =>
+            replaceDelegateCalls(
+              chain,
+              MultiAddress.Id(proxy),
+              configurator.address,
+              MultiAddress.Id(members[i]!),
+            )
+          ),
+        ])
+      )
+      .into(MetaRune)
+      .flat()
+    const ownershipSwaps = chain
+      .extrinsic(Rune.rec({
+        type: "Utility",
+        value: Rune.rec({
+          type: "batchAll",
+          calls: ownershipSwapCalls,
+        }),
+      }))
+      .signed({ sender: configurator })
+      .sent()
+      .dbgStatus("Ownership swaps:")
+      .finalized()
+
+    return proxies
+      .chain(() => existentialDeposits)
+      .chain(() => ownershipSwaps)
+      .chain(() =>
+        Rune.rec({
+          signatories: memberProxies,
+          threshold,
+          stash: stashProxy,
+        })
+      )
+      .into(VirtualMultisigRune, chain)
+  }
+}
+
+export interface VirtualMultisigDeploymentProps extends Multisig {
+  deployer: ExtrinsicSender
+  deposits?: bigint
+}
+
+export function filterPureCreatedEvents<X>(...[events]: RunicArgs<X, [Event[]]>) {
+  return Rune.resolve(events).map((events) =>
+    events
+      .map((e) => e.event)
+      .filter((event): event is RuntimeEvent.Proxy => event.type === "Proxy")
```

Comment:
Dont we have codegen guard for these?

Reply:
We should avoid using codegened runtime code in patterns, as it makes them unusable to consuming NodeJS projects (for now).